### PR TITLE
feat(compat) preload Node.js native modules in global vars in node compat REPL

### LINKED ISSUE #13109: compat: preload Node.js native modules in global vars in node compat REPL
In Node.js, all native modules are available as global variables in REPL:

```shellsession
$ node
Welcome to Node.js v16.9.1.
Type ".help" for more information.
> util
{
  _errnoException: [Function: __node_internal_errnoException],
  _exceptionWithHostPort: [Function: __node_internal_exceptionWithHostPort],
  _extend: [Function: _extend],
  callbackify: [Function: callbackify],
  ...
```

How about doing the same thing with `deno repl --compat --unstable`?

### PR DESCRIPTION
**Pull request by VishnuJin**
fixes #13109

### CODE REVIEW

File: cli/main.rs
Comment (bartlomieju) [inline cli/main.rs:965]:
```suggestion
    worker.run_event_loop(false).await?;
    compat::setup_builtin_modules(&mut worker.js_runtime)?;
```

File: cli/compat/mod.rs
Comment (VishnuJin) [inline cli/compat/mod.rs:135]: require or globalThis.require throws `error: Uncaught ReferenceError: require is not defined`

Comment (bartlomieju) [reply cli/compat/mod.rs:135]: Apply suggestion from my previous comment to fix this problem

File: cli/compat/mod.rs
Comment (VishnuJin) [inline cli/compat/mod.rs:26]: @bartlomieju the commented modules are actually not resolved and throws Module Not found error, how do we want to handle this ?

File: cli/compat/mod.rs
Comment (VishnuJin) [inline cli/compat/mod.rs:32]: deprecated from node https://nodejs.org/api/domain.html

**Comment by VishnuJin** [comment]
The following modules are stated as unsupported in this [README](https://github.com/denoland/deno_std/blob/e59b24543e670752701cd6b7d35ebb857d100210/node/README.md) 
Should they be removed from `SUPPORTED_MODULES` array ? or perhaps moved into `UNSUPPORTED_MODULES` array ?
- cluster
- domain (deprecated from Node)
- https
- tls
- v8

**Comment by bartlomieju** [comment]
@VishnuJin I believe we will just mock out APIs for these modules in `deno_std` and use the full list in `cli/compat/mod.rs`.

**Comment by bartlomieju** [comment]
https://github.com/denoland/deno_std/pull/1737 will unblock this PR, but we will have to wait until next release to actually use it.

**Comment by bartlomieju** [comment]
@VishnuJin `deno_std` was update, you can merge `main` branch and proceed with the PR

**Comment by VishnuJin** [comment]
@bartlomieju  thanks for the update!!I'm on vacation and I'll be back by next Wednesday and will resume working with this PR

**Comment by bartlomieju** [comment]
@VishnuJin let me know if you need more pointers for this PR

**Comment by VishnuJin** [comment]
@bartlomieju I see the deno_std git submodule in Deno main is not pointing to 119.0 and after updating it to 119.0 the [test fails](https://github.com/denoland/deno/runs/4673672317?check_suite_focus=true#step:34:1580) for which I think the tests needs to be updated !!

**Comment by bartlomieju** [comment]
@VishnuJin you should be updating `test_util/` directories. Everything that you need to finish this PR has already been updated. Please revert changes to `test_util/` submodules.

**Comment by VishnuJin** [comment]
@bartlomieju sorry I'm still under confusion 😟

- currently I see, under `test_utils/` dir `std @ e59b245` points to older version and hence the tests can't load modules
image

- also in the newer version i.e `0.119` the file `server_legacy.ts` has been removed which is required by integration test `045_proxy_test.ts` which causes the [integration test to fail](https://github.com/denoland/deno/runs/4673672317?check_suite_focus=true#step:34:1580)

**Comment by bartlomieju** [comment]
@VishnuJin these things are unrelated to this PR - in "compat" mode files are pulled from `deno.land/std`, it's defined here: https://github.com/denoland/deno/blob/003534e8836043892499bac86424e059fc14c2f1/cli/compat/mod.rs#L19

Looks like `async_hooks` module is actually not available. I will fix that after NYE

**Comment by VishnuJin** [comment]
@bartlomieju In that case I guess the following modules also needs to be checked as they are also throwing same error
- async_hooks
- cluster
- dgram
- domain
- https
- tls
- v8

**Comment by bartlomieju** [comment]
@VishnuJin sorry for misunderstanding; I forgot that compat tests actually use local git submodule of `deno_std` to speed up tests. I will handle the upgrade of the submodule in #13260 and let you know once that's ready.

**Comment by VishnuJin** [comment]
@bartlomieju no problem 🙂

**Comment by bartlomieju** [comment]
@VishnuJin you can now merge `main` branch into your PR and run `git submodule update --init --recursive` to make your PR work.

File: cli/tests/integration/compat_tests.rs
Comment (VishnuJin) [inline cli/tests/integration/compat_tests.rs:137]: @bartlomieju is this test enough ?

Comment (bartlomieju) [reply cli/tests/integration/compat_tests.rs:137]: Yes, it's fine, thanks

File: cli/compat/mod.rs
Comment (bartlomieju) [inline cli/compat/mod.rs:153]: Let's rewrite this for loop to create a single script and execute it only once. Also I suggest to rename the script to `setup_node_builtins.js`

**Review by bartlomieju — CHANGES_REQUESTED** [review]
@VishnuJin one last nitpick and we're ready to land this PR

File: cli/compat/mod.rs
Comment (VishnuJin) [reply cli/compat/mod.rs]: updated and squashed to single commit

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you @VishnuJin

**Pull request merged**
